### PR TITLE
Allow private key and nore url only in the env file

### LINKED ISSUE #300: Remove public/private key entries from the config.
At the moment can set priv/pub key in the config.json and the .env file. In the past people would paste their config file exposing the private key so this led to moving these in the `.env` file, but one can still set these up in the config.json as well. Lets change it to allow only using the .env file and remove these from the config.json file.

### LINKED ISSUE #313: Move the infura API key to the .env file so that people providing their configs don't expose it accidentally.

### PR DESCRIPTION
**Pull request by themandalore**
fixes #300 
fixes #313

### CODE REVIEW

**Comment by themandalore** [comment]
any idea on this last test @krasi-georgiev ?

File: pkg/config/testing.go [hunk 1]
```diff
@@ -16,17 +16,16 @@ import (
...
+    "publicAddress": "92f91500e105e3051f3cf94616831b58f6bce1e8",
     "serverHost": "localhost",
     "serverPort": 5000,
     "trackerCycle": 1,
     "trackers": {},
     "dbFile": "/tellorDB",
     "requestTips": 1,
-    "configFolder": "` + filepath.Join("..", "..", "configs") + `"
+    "configFolder": "` + filepath.Join("..", "..", "configs") + `",
+    "envFile": "` + filepath.Join("..", "..", "configs/.env.example") + `"
```
Comment (krasi-georgiev) [inline pkg/config/testing.go:28]:
```suggestion
    "envFile": "` + filepath.Join("..", "..", "configs",".env.example") + `"
```

The idea with `filepath.Join` is that under Linux the delimiter is `/` and under windows it uses `\`

File: pkg/db/remoteRequest_test.go [hunk 2]
```diff
@@ -18,7 +18,7 @@ import (
 
 func TestRemoteRequestCodec(t *testing.T) {
 	cfg := config.OpenTestConfig(t)
-	cfg.ServerWhitelist = []string{"0x053b09e98ede40997546e8bb812cd838f18bb146"}
+	cfg.ServerWhitelist = []string{"0x92f91500e105e3051f3cf94616831b58f6bce1e8"}
```
Comment (krasi-georgiev) [inline pkg/db/remoteRequest_test.go:21]: Why this needed changing?

File: pkg/pow/miningGroup_test.go [hunk 3]
```diff
@@ -187,11 +187,18 @@ func BenchmarkHashFunction(b *testing.B) {
...
+    "databaseURL": "http://localhost7545",
+    "publicAddress": "92f91500e105e3051f3cf94616831b58f6bce1e8",
+    "serverHost": "localhost",
+    "serverPort": 5000,
+    "trackerCycle": 1,
+    "trackers": {},
+    "dbFile": "/tellorDB",
+    "requestTips": 1,
+    "configFolder": "` + filepath.Join("..", "..", "configs") + `",
+    "envFile": "` + filepath.Join("..", "..", "configs/.env.example") + `"
```
Comment (krasi-georgiev) [inline pkg/pow/miningGroup_test.go:200]:
```suggestion
    "envFile": "` + filepath.Join("..", "..", "configs",".env.example") + `"
```

File: pkg/tracker/main_test.go [hunk 4]
```diff
@@ -16,13 +17,17 @@ import (
...
+    "databaseURL": "http://localhost7545",
+    "publicAddress": "92f91500e105e3051f3cf94616831b58f6bce1e8",
+    "serverHost": "localhost",
+    "serverPort": 5000,
+    "trackerCycle": 1,
+    "trackers": {},
+    "dbFile": "/tellorDB",
+    "requestTips": 1,
+    "configFolder": "` + filepath.Join("..", "..", "configs") + `",
+    "envFile": "` + filepath.Join("..", "..", "configs/.env.example") + `"
```
Comment (krasi-georgiev) [inline pkg/tracker/main_test.go:30]:
```suggestion
    "envFile": "` + filepath.Join("..", "..", "configs",".env.example") + `"
```

**Review by krasi-georgiev — COMMENTED** [review]
Lets add the an entry in the gitignore file to not add files like:
`pkg/tracker/possible-dispute-2020-12-10 15:22:10 -0500 EST.txt`

Can't think of a reason why the tests are failing, but ping me if you can't figure it out and will have a look.

This change is also relevant to the end-users so let's also add an entry in the changelog.

This part also needs to be updated in the docs.

```
sed -i -e '1,/0xE037EC8EC9ec423826750853899394dE7F024fee/ s/0xE037EC8EC9ec423826750853899394dE7F024fee/0xcdd8FA31AF8475574B8909F135d510579a8087d3/' config2.json
sed -i -e '1,/0xE037EC8EC9ec423826750853899394dE7F024fee/ s/0xE037EC8EC9ec423826750853899394dE7F024fee/0xb9dD5AfD86547Df817DA2d0Fb89334A6F8eDd891/' config3.json
sed -i -e '1,/0xE037EC8EC9ec423826750853899394dE7F024fee/ s/0xE037EC8EC9ec423826750853899394dE7F024fee/0x230570cD052f40E14C14a81038c6f3aa685d712B/' config4.json
sed -i -e '1,/0xE037EC8EC9ec423826750853899394dE7F024fee/ s/0xE037EC8EC9ec423826750853899394dE7F024fee/0x3233afA02644CCd048587F8ba6e99b3C00A34DcC/' config5.json
```
Create `.env1` file with the private key for each miner.
```text
echo "4bdc16637633fa4b4854670fbb83fa254756798009f52a1d3add27fb5f5a8e16" > .env1
echo "d32132133e03be292495035cf32e0e2ce0227728ff7ec4ef5d47ec95097ceeed" > .env2
echo "d13dc98a245bd29193d5b41203a1d3a4ae564257d60e00d6f68d120ef6b796c5" > .env3
echo "4beaa6653cdcacc36e3c400ce286f2aefd59e2642c2f7f29804708a434dd7dbe" > .env4
echo "78c1c7e40057ea22a36a0185380ce04ba4f333919d1c5e2effaf0ae8d6431f14" > .env5
```

File: pkg/db/remoteRequest_test.go (hunk 2 above)
Comment (themandalore) [reply pkg/db/remoteRequest_test.go:21]: Changed the private key for some reason (there was a reason, I just forget it now)

**Comment by themandalore** [comment]
should be good now @krasi-georgiev

File: docs/setup-and-usage.md [hunk 5]
```diff
@@ -281,14 +280,18 @@ sed -i -e '1,/0xE037EC8EC9ec423826750853899394dE7F024fee/ s/0xE037EC8EC9ec423826
...
-echo "78c1c7e40057ea22a36a0185380ce04ba4f333919d1c5e2effaf0ae8d6431f14" > .env5
+SAMPLE ETH_PRIVATE_KEY(S):
+"4bdc16637633fa4b4854670fbb83fa254756798009f52a1d3add27fb5f5a8e16" > .env1
+"d32132133e03be292495035cf32e0e2ce0227728ff7ec4ef5d47ec95097ceeed" > .env2
+"d13dc98a245bd29193d5b41203a1d3a4ae564257d60e00d6f68d120ef6b796c5" > .env3
+"4beaa6653cdcacc36e3c400ce286f2aefd59e2642c2f7f29804708a434dd7dbe" > .env4
+"78c1c7e40057ea22a36a0185380ce04ba4f333919d1c5e2effaf0ae8d6431f14" > .env5
+
+SAMPLE NODE_URL
+"https://mainnet.infura.io/v3/xxxxxxxxxxxxx"
```
Comment (krasi-georgiev) [inline docs/setup-and-usage.md:294]:
```suggestion
echo "ETH_PRIVATE_KEY=4bdc16637633fa4b4854670fbb83fa254756798009f52a1d3add27fb5f5a8e16" > .env1
echo "ETH_PRIVATE_KEY=d32132133e03be292495035cf32e0e2ce0227728ff7ec4ef5d47ec95097ceeed" > .env2
echo "ETH_PRIVATE_KEY=d13dc98a245bd29193d5b41203a1d3a4ae564257d60e00d6f68d120ef6b796c5" > .env3
echo "ETH_PRIVATE_KEY=4beaa6653cdcacc36e3c400ce286f2aefd59e2642c2f7f29804708a434dd7dbe" > .env4
echo "ETH_PRIVATE_KEY=78c1c7e40057ea22a36a0185380ce04ba4f333919d1c5e2effaf0ae8d6431f14" > .env5


echo "NODE_URL=https://mainnet.infura.io/v3/xxxxxxxxxxxxx" >> .env1
echo "NODE_URL=https://mainnet.infura.io/v3/xxxxxxxxxxxxx" >> .env2
echo "NODE_URL=https://mainnet.infura.io/v3/xxxxxxxxxxxxx" >> .env3
echo "NODE_URL=https://mainnet.infura.io/v3/xxxxxxxxxxxxx" >> .env4
echo "NODE_URL=https://mainnet.infura.io/v3/xxxxxxxxxxxxx" >> .env5
```

Do we want this to be an exact copy and paste command?

**Review by krasi-georgiev — APPROVED** [review]
Just one small nit(see my comment) with the configs and once fixed feel free to merge.
Thanks

File: docs/changelog.md [hunk 6]
```diff
@@ -10,6 +10,8 @@ We use _breaking :warning:_ to mark changes that are not backward compatible \(r
 
 ## Unreleased
 
+*[\#317](https://github.com/tellor-io/telliot/pull/317) Removed nodeURL and private key from config file
```
Comment (krasi-georgiev) [inline docs/changelog.md:13]:
```suggestion
* [\#317](https://github.com/tellor-io/telliot/pull/317) Removed nodeURL and private key from config file
```

Comment (krasi-georgiev) [reply docs/changelog.md:13]: I am not sure without this space if it will show properly in gitbook

**Comment by krasi-georgiev** [comment]
@themandalore thanks!

**Pull request merged**